extensions/email/src/main/java/org/mule/extension/email/api/EmailContentProcessor.java

```diff
@@ -58,84 +63,91 @@ private EmailContentProcessor(javax.mail.Message message) {
-      }
-    } catch (MessagingException | IOException e) {
-      throw new EmailException(ERROR_PROCESSING_MESSAGE, e);
-    }
-  }
-
-  /**
    * Processes a single {@link Part} and adds it to the body of the message or as a new attachment depending on it's disposition
    * type.
    *
    * @param part the part to be processed
    */
-  private void processPart(Part part) {
+  private void processPart(Part part, StringJoiner bodyCollector) {
     try {
       Object content = part.getContent();
       if (isMultipart(content)) {
-        processMultipartPart(part);
+        Multipart mp = (Multipart) part.getContent();
+        for (int i = 0; i < mp.getCount(); i++) {
+          processPart(mp.getBodyPart(i), bodyCollector);
+        }
       }
 
       if (isAttachment(part)) {
-        Map<String, LinkedList<String>> headers = new HashMap<>();
-        final Enumeration allHeaders = part.getAllHeaders();
-        while (allHeaders.hasMoreElements()) {
-          Header h = (Header) allHeaders.nextElement();
-          headers.put(h.getName(), new LinkedList<>(singletonList(h.getValue())));
-        }
-
-        attachmentParts.add(Message.builder().payload(part.getInputStream()).mediaType(MediaType.parse(part.getContentType()))
-            .attributes(new PartAttributes(part.getFileName(), part.getFileName(), part.getSize(), headers)).build());
+        processAttachment(part);
       } else {
-        if (isText(content)) {
-          body.add((String) content);
-        }
-
-        if (content instanceof InputStream && isInline(part) && part.isMimeType(TEXT)) {
-          String inline = IOUtils.toString((InputStream) content);
-          body.add(inline);
-        }
+        processBodyPart(part, bodyCollector, content);
       }
     } catch (MessagingException | IOException e) {
       throw new EmailException(ERROR_PROCESSING_MESSAGE, e);
     }
   }
 
   /**
+   * Processes a body part, adding it to the body that is being built.
+   *
+   * @param part the attachment part to be processed
+   */
+  private void processBodyPart(Part part, StringJoiner bodyCollector, Object content) throws MessagingException {
+    if (isText(content)) {
+      bodyCollector.add((String) content);
+    }
+
+    if (content instanceof InputStream && isInline(part) && part.isMimeType(TEXT)) {
+      String inline = IOUtils.toString((InputStream) content);
+      bodyCollector.add(inline);
+    }
+  }
+
+  /**
+   * Processes an attachment part, adding it to the attachments list.
+   *
+   * @param part the attachment part to be processed
+   */
+  private void processAttachment(Part part) throws MessagingException, IOException {
+    Map<String, LinkedList<String>> headers = new HashMap<>();
+    final Enumeration allHeaders = part.getAllHeaders();
+    while (allHeaders.hasMoreElements()) {
+      Header h = (Header) allHeaders.nextElement();
+      headers.put(h.getName(), new LinkedList<>(singletonList(h.getValue())));
```

Comment:
A non mutable copy of an immutable single element list... overkill... Just use Arrays.asList() instead since that list will never be exposed to the user anyway